backend/core/views.py

```
@@ -4148,6 +4147,84 @@ def compliance_assessment_csv(self, request, pk):
 
+    @action(detail=True, methods=["get"], name="Audit as an Excel")
+    def xlsx(self, request, pk):
+        (viewable_objects, _, _) = RoleAssignment.get_accessible_object_ids(
+            Folder.get_root_folder(), request.user, ComplianceAssessment
+        )
+        if UUID(pk) not in viewable_objects:
+            return Response(
+                {"error": "Permission denied"}, status=status.HTTP_403_FORBIDDEN
+            )
+        audit = ComplianceAssessment.objects.get(id=pk)
+        req_nodes = RequirementNode.objects.filter(framework=audit.framework)
```

Ruff (0.11.9)

`RequirementNode` may be undefined, or defined from star imports

(F405)